### PR TITLE
Pin app.kubernetes.io/name label to use .Chart.Name

### PR DESCRIPTION
**Pull request by ubergesundheit**
This PR changes the `app.kubernernetes.io/name` label to be more stable.

In regards to monitoring, this will help us to identify kong-app deployments regardless of users setting `.Values.nameOverride`

### CODE REVIEW

**Comment by webwurst** [comment]
Does the `nameOverride` still show up as a label? To distinct different deployments. Is it maybe this? `app.kubernetes.io/instance: "{{ .Release.Name }}"`

**Comment by ubergesundheit** [comment]
> Does the nameOverride still show up as a label? To distinct different deployments. Is it maybe this? app.kubernetes.io/instance: "{{ .Release.Name }}"

Yes, the `app.kubernetes.io/instance` label uses the `kong.name` helper which uses the nameOverride instead of .Chart.Name if set

**Comment by Gacko** [comment]
The whole idea basically looks fine to me. But I'm not sure if it's the correct way to implement it, yet. Plus: We should maybe use a more generic name describing what we'd like to achieve so anyone reading this template understands why we're not using the normal template. Just my humble opinion.

**Comment by ubergesundheit** [comment]
> I'm not sure if it's the correct way to implement it, yet.

Any more concrete suggestions?

> We should maybe use a more generic name describing what we'd like to achieve so anyone reading this template understands why we're not using the normal template.

Can you quickly add a note on the specific line you're referring to? I've added a comment in 807f32b

**Comment by Gacko** [comment]
Perfect! :)

**Review by Gacko — APPROVED** [review]

File: helm/kong-app/templates/_helpers.tpl [hunk 1]
```diff
@@ -12,6 +12,10 @@ We truncate at 63 chars because some Kubernetes name fields are limited to this
 {{- default .Chart.Name .Values.nameOverride | trunc 63 | trimSuffix "-" -}}
 {{- end -}}
 
+{{- define "kong.app-kubernetes-io-name" -}}
+{{- .Chart.Name | trunc 63 | trimSuffix "-" | quote -}}
```
[the remark this answers is not in the document]
Comment (ubergesundheit) [reply helm/kong-app/templates/_helpers.tpl:20]: Good point. I'll add a follow up PR

**Pull request merged**